### PR TITLE
feat(endpoint): create movie plugin, routes, and validators

### PR DESCRIPTION
**Pull request by Lianathanoj**
### Context:
This is for the __Endpoint Configuration, Validators, Validator Unit Tests, and Controllers__ portion of Miyagi. Before this is completed, we need to have completed the __Model Unit Tests__ portion. Creating the POST /movie endpoint will allow us to create a movie in the database via the controller, and creating the validator will allow us to set constraints for the movie attributes.

### What:
* Created the feature plugin for movie which configures routing and the response for hitting that particular POST endpoint. 
* Created a validator for the movie with specific constraints on the title and release year.
* Set a more specific error response code (from 400 to 422).
* Create a controller to place the response from the POST request into the database.
* Created tests for the validator.

### Why:
* We want to be able to add movies to the database
* We want to make sure that the data we pass to an endpoint is valid
* We need the endpoint response to be more specific

### CODE REVIEW

File: lib/plugins/features/movies/index.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+const MovieValidator = require('../../../validators/movie');
+
+exports.register = (server, options, next) => {
+  server.route([{
+    method: 'POST',
+    path: '/movies',
+    config: {
+      handler: (request, reply) => {
+        reply('hello world');
```
Comment (sjl2) [inline lib/plugins/features/movies/index.js]: Is coverage passing for this line? What test is covering this?

**Review by sjl2 — APPROVED** [review]
LGTM

File: lib/plugins/features/movies/index.js (hunk 1 above)
Comment (kienan) [reply lib/plugins/features/movies/index.js]: @Lianathanoj Related to this, what are your thoughts on going ahead and implementing the controller here?  We probably don't want our endpoint available but only returning `hello world`.

If you did want to split up this PR, I would recommend perhaps the first being just the validator and then the second implementing the endpoint with controller included. Let me know if you have any questions!

Comment (Lianathanoj) [reply lib/plugins/features/movies/index.js]: If I understand this part correctly, I believe that the validator tests that I wrote create a payload mimicking a request with certain JSON parameters. We then validate the payload itself. The handler which replies to a request with a status code and response of 'hello world' is mainly just used to make sure that the endpoint even works. 

When I make another PR for controllers (actually updating the database after the POST request has gone through), these lines of code will change.

EDIT: Just saw kienan's comment. I'll implement the controller here.

**Review by kienan — APPROVED** [review]
Noice! LGTM, though can you remove the empty `authentication.js` file?

**Pull request merged**